drivers/virtio/virtio-serial.c

```diff
@@ -523,6 +523,52 @@ static void virtio_serial_uninit(FAR struct virtio_serial_priv_s *priv)
   virtio_free_buf(vdev, priv->udev.recv.buffer);
 }
 
+/****************************************************************************
+ * Name: virtio_serial_uart_register
+ ****************************************************************************/
+
+static int virtio_serial_uart_register(FAR struct virtio_serial_priv_s *priv)
+{
+  FAR const char *name = CONFIG_DRIVERS_VIRTIO_SERIAL_NAME;
+  bool found = false;
+  int start = 0;
+  int ret;
+  int i;
+  int j;
+
+  for (i = 0, j = 0; name[start] != '\0'; i++)
+    {
+      if (name[i] == ';' || name[i] == '\0')
+        {
+          if (j++ == g_virtio_serial_idx)
+            {
+              found = true;
+              break;
+            }
+
+          start = i + 1;
+        }
+    }
+
+  if (found)
+    {
+      snprintf(priv->name, NAME_MAX, "/dev/%.*s", i - start, &name[start]);
+    }
+  else
+    {
+      snprintf(priv->name, NAME_MAX, "/dev/ttyV%d", g_virtio_serial_idx);
+    }
+
+  ret = uart_register(priv->name, &priv->udev);
```

Comment:
```suggestion
  g_virtio_serial_idx++;
  return uart_register(priv->name, &priv->udev);
}
```
increase index before register uart to skip the invalid name